src/EventStore.Core/LogV3/LogV3RecordFactory.cs

```diff
@@ -73,5 +77,47 @@ public class LogV3RecordFactory : IRecordFactory<StreamId> {
 				metadata: metadata.Span);
 			return result;
 		}
+
+		public ILogRecord CreatePartitionTypeRecord(
+			DateTime timeStamp,
+			long logPosition,
+			Guid partitionTypeId,
+			Guid partitionId,
+			string name) {
+			
+			return new PartitionTypeLogRecord(
+				timeStamp,
```

Comment:
personally i'd add the argument names here, but up to you. just so when reading the code i dont wonder 'is there any chance the guids are back to front'